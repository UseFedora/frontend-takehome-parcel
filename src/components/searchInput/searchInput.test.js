import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from './';

describe('SearchInput', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchInput />);
    expect(component).toMatchSnapshot();
  });

  it ('should display the placeholder text', () => {
    const component = shallow(
      <SearchInput
        placeholderText="Enter search value here"
      />
    );
    expect(component.html()).toContain('Enter search value here');
  });

  it ('should display the search value text', () => {
    const component = shallow(
      <SearchInput
        searchValue="railties"
      />
    );
    expect(component.html()).toContain('railties');
  });

  it ('should invoke the submit handler if the user presses <ENTER>', () => {
    const spyOnSubmit = jest.fn();
    const component = shallow(
      <SearchInput
        onSubmit={spyOnSubmit}
      />
    );
    component.find('input').simulate('keypress', { key: 'Enter' });
    expect(spyOnSubmit).toHaveBeenCalled();
  });

  it ('should not invoke the submit handler if another key is pressed', () => {
    const spyOnSubmit = jest.fn();
    const component = shallow(
      <SearchInput
        onSubmit={spyOnSubmit}
      />
    );
    component.find('input').simulate('keypress', { key: 'Tab' });
    expect(spyOnSubmit).not.toHaveBeenCalled();
  });

  it ('should call the change handler with the supplied value', () => {
    const spyOnChange = jest.fn();
    const component = shallow(
      <SearchInput
        onChange={spyOnChange}
      />
    );
    component.find('input').simulate('change', { target: { value: 'rails' } });
    expect(spyOnChange).toHaveBeenCalledWith('rails');
  });
});
