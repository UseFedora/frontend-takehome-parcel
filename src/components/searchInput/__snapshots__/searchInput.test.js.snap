// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput
    onChange={[Function]}
    onSubmit={[Function]}
    placeholderText=""
    searchValue=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder=""
        type="text"
        value=""
      />,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onChange": [Function],
        "onKeyPress": [Function],
        "placeholder": "",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder=""
          type="text"
          value=""
        />,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "onChange": [Function],
          "onKeyPress": [Function],
          "placeholder": "",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
